### PR TITLE
Field name attribute documentation 

### PR DESCRIPTION
**Pull request by rosswbrown**
Modular connection dialog field name attribute documentation.

This is the minimal set of name and value field information for platform integration:
- Endpoint
- SSL
- Authentication

This is the most concise way I found to communicate the core platform requirements.  I did not try to document the entire set of reserved connection attributes and values that was compiled here: https://gitlab.tableausoftware.com/connectivity/connector-plugin-sdk-internal/blob/master/docs/attributes.md

I'd be open to people's thoughts on the best way to communicate the entire set of reserved attributes.  I'll open anything like that in a follow-up change.

### CODE REVIEW

File: docs/mcd.md [hunk 1]
```diff
@@ -182,8 +225,6 @@ The images show the Connection Dialog produced using the Connection Fields file
     </conditions>
   </field>
  
-  <field name="AutoReconnect" label="" value-type="string" default-value="0" editable="false" />
```
Comment (rosswbrown) [inline docs/mcd.md]: This was intentional removal

Comment (sxlin) [reply docs/mcd.md]: It would be great if you can add sslmode in the example.

Comment (rosswbrown) [reply docs/mcd.md]: I'll add a note to do as follow-up in the TFS item.  I'll need to update the screenshots if I change examples here.

Comment (rosswbrown) [reply docs/mcd.md]: Instead of modifying existing example I opted to make a clean boolean focused Example 3.

File: docs/mcd.md [hunk 2]
```diff
@@ -114,8 +114,51 @@ The connector will not load if there are circular references.
 | field | Name of the field whose value will be checked for equality | No | Any string value | |
 | value | The value to match. If the field's value is equal to this, the condition is true. | No | Any string value | |
 
+## Connection Field Platform Integration 
 
-## Example - A Non-Editable Field
+As referenced above in the `<field>` element section, some `name` attribute values describe platform functionality. It is important that if the connector functionality desired matches any of the descriptions below then names and values below must be used.
```
Comment (joe-langford) [inline docs/mcd.md]: nits: consider 
"connector functionality desired" -> "desired connector functionality"
"then names" -> "then the names"

File: docs/mcd.md [hunk 3]
```diff
@@ -114,8 +114,51 @@ The connector will not load if there are circular references.
...
+
+Additionally there are a set of reserved `name` attribute values not documented at this time. Recommendations, documentation and enforcement coming soon.
+
+### Endpoint
+
+The endpoint attributes describe the unique parameters of a connection. Many connections provider additional field names and values not defined by the platform.
```
Comment (joe-langford) [inline docs/mcd.md]: provider -> provide

**Pull request merged**